Android TV Remote integration documentation (#26659)
home-assistant/home-assistant.github.io: source/_integrations/androidtv_remote.markdown

---
title: Android TV Remote
description: Instructions on how to integrate Android TV remotes into Home Assistant.
ha_category:
  - Remote
ha_release: 2023.4
ha_iot_class: Local Push
ha_config_flow: true
ha_codeowners:
  - '@tronikos'
ha_quality_scale: platinum
ha_domain: androidtv_remote
ha_zeroconf: true
ha_platforms:
  - diagnostics
  - remote
ha_integration_type: device
---

The `androidtv_remote` platform allows you to control an Android TV device by sending [commands](https://github.com/tronikos/androidtvremote2/blob/main/TvKeys.txt) and launching apps.

{% include integrations/config_flow.md %}

## Entity

> This integration adds a `remote` entity that if you toggle it, it turns on/off the Android TV device.

Reviewer: ```suggestion
This integration adds a `remote` entity which turns on/off the Android TV device.
```

---
title: Android TV Remote
description: Instructions on how to integrate Android TV remotes into Home Assistant.
ha_category:
  - Remote
ha_release: 2023.4
ha_iot_class: Local Push
ha_config_flow: true
ha_codeowners:
  - '@tronikos'
ha_quality_scale: platinum
ha_domain: androidtv_remote
ha_zeroconf: true
ha_platforms:
  - diagnostics
  - remote
ha_integration_type: device
---

> The `androidtv_remote` platform allows you to control an Android TV device by sending [commands](https://github.com/tronikos/androidtvremote2/blob/main/TvKeys.txt) and launching apps.

Reviewer 2: It isn't a platform?

```suggestion
The Android TV Remote integration allows you to control an Android TV device by sending [commands](https://github.com/tronikos/androidtvremote2/blob/main/TvKeys.txt) and launching apps.
```

Author: You are right. I copied this from https://www.home-assistant.io/integrations/harmony/ Fixing that one too in https://github.com/home-assistant/home-assistant.io/pull/26750

{% include integrations/config_flow.md %}

## Entity

This integration adds a `remote` entity which turns on/off the Android TV device.
The entity has the `current_activity` attribute that shows the current foreground app on the Android TV.

## Services

> ### Service `remote.turn_off`

Reviewer 2: This isn't a service provided by this integration, but by the `remote` integration.
Let's remove it here, as we do not need to create duplicate documentation.

Author: I followed the documentation for other remote integrations which mention these services
Examples:
https://www.home-assistant.io/integrations/harmony/
https://www.home-assistant.io/integrations/apple_tv/#remote
https://www.home-assistant.io/integrations/xbox/#remote

Author: Maybe remove the attributes and the documentation for the services and just have examples for each service? All the other integrations I linked to in my previous comment list the service attributes though. I see some value in that since not all integrations support all the attributes. e.g. I think only one other integration supports the hold_secs attribute of send_command. My integration supports all the attributes of send_command but then it doesn't support learn_command. So I think there is value explicitly listing all the supported services and how to call them.
I also just noticed the documentation at https://www.home-assistant.io/integrations/remote/ doesn't have good documentation for the services. Maybe that's intended since not all remote integrations support them and it's better to have each implementation list the supported ones.

Author: In addition there is value explaining what each attribute is since it might be slightly different between remote implementations. e.g. Harmony for activity uses presets e.g. Watch TV that can send a series of signals to different devices. In my integration that's app to open, e.g. YouTube.

Reviewer 2: We should not duplicate documentation. It is problematic to keep in sync.

> I also just noticed the documentation at [home-assistant.io/integrations/remote](https://www.home-assistant.io/integrations/remote/) doesn't have good documentation for the services.

Feel free to expand that page in a separate PR.

Author: I kept the sections of the services, removed the tables and kept the examples as I think they are useful.

Reviewer 2: I disagree, most certainly with a simple turn on as this one. If you want to provide additional examples with specific use cases, I think that is fine to create (as general examples).

But we should not re-document services from other integrations.

Author: I removed the section headers, the 1 sentence explanation of each service, and the simpler turn off and on examples.

Turn off the Android TV.

| Service data attribute | Optional | Description |
| ---------------------- | -------- | ----------- |
| `entity_id`            |       no | Entity ID to target.

> ### Service `remote.turn_on`

Reviewer 2: This isn't a service provided by this integration, but by the `remote` integration.
Let's remove it here, as we do not need to create duplicate documentation.

Author: Resolving and keeping the other comment thread open. Depending on feedback there I can do the same for these sections too.

Turn on the Android TV. If `activity` is specified it will open the specified URL in the associated app.

| Service data attribute | Optional | Description |
| ---------------------- | -------- | ----------- |
| `entity_id`            |       no | Entity ID to target.
| `activity`             |      yes | URL to open.

Examples of URLs:

| App | URL |
| --- | --- |
| YouTube | https://www.youtube.com
| Netflix | https://www.netflix.com/title
| Prime Video | https://app.primevideo.com
| Disney+ | https://www.disneyplus.com

Example for launching YouTube:

```yaml
service: remote.turn_on
data:
  activity: https://www.youtube.com
target:
  entity_id: remote.living_room_tv
```

Example for opening a specific YouTube video:

```yaml
service: remote.turn_on
data:
  activity: https://www.youtube.com/watch?v=dQw4w9WgXcQ
target:
  entity_id: remote.living_room_tv
```

> ### Service `remote.send_command`

Reviewer 2: This isn't a service provided by this integration, but by the `remote` integration.
Let's remove it here, as we do not need to create duplicate documentation.

| `num_repeats`          |      yes | The number of times you want to repeat the command(s).
| `delay_secs`           |      yes | The time you want to wait in between repeated commands.
| `hold_secs`            |      yes | The time you want to have it held before the release is send.

Example for opening the currently selected item on the Android TV:

```yaml
action: call-service
service: remote.send_command
data:
  command: DPAD_CENTER
target:
  entity_id: remote.living_room_tv
```

Example for long pressing on the currently selected item on the Android TV:

```yaml
action: call-service
service: remote.send_command
data:
  command: DPAD_CENTER
  hold_secs: 0.5
target:
  entity_id: remote.living_room_tv
```

## Dashboard

You have to manually create buttons in Lovelace to send commands to the Android TV device or launch apps on it.

Below is an example for you to start with. Many of the buttons support long press.
Replace all instances of `living_room_tv` with your entity ID.

![Screenshot Android TV Remote example](/images/integrations/androidtv_remote/lovelace_example.png)

```yaml
type: vertical-stack
cards:

>   - type: entities

Reviewer 2: This is quite a huge piece of YAML in this documentation, and maybe a bit off-topic? Let's remove it here and feel free to create a community guide for it. A smaller or simpler example would be acceptable IMHO.

Author: I followed the example in https://www.home-assistant.io/integrations/xbox/#picture-elements-card which also has YAML code. Would it help if I have it initially collapsed and let users expand it?

Reviewer 2: Sure, that works 👍

Author: Done.

target:
  entity_id: remote.living_room_tv
```

Examples of URLs to pass as activity for some popular apps:

| App | URL |
| --- | --- |
| YouTube | https://www.youtube.com
| Netflix | https://www.netflix.com/title
| Prime Video | https://app.primevideo.com
| Disney+ | https://www.disneyplus.com

### Service `remote.send_command`

Send a single command or a set of commands to the Android TV.
You can optionally specify the number of times you wish to repeat the command(s), delay you want between repeated command(s), and hold.

For a list of the most common commands you can send to the Android TV see: [TvKeys](https://github.com/tronikos/androidtvremote2/blob/main/TvKeys.txt).
For a full list see [here](https://github.com/tronikos/androidtvremote2/blob/main/src/androidtvremote2/remotemessage.proto#L90).

Examples:

```yaml
# Open the currently selected item on the Android TV
service: remote.send_command
data:
  command: DPAD_CENTER
target:
  entity_id: remote.living_room_tv
```

```yaml
# Long press on the currently selected item on the Android TV
service: remote.send_command
data:
  command: DPAD_CENTER
  hold_secs: 0.5
target:
  entity_id: remote.living_room_tv
```

> ## Dashboard

Reviewer 2: ```suggestion
## Dashboard example
```
